components/blocks/MediaComponent.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import Image from 'next/image';
+
+const MediaComponent = ({ data }) => {
+  const { headline, mediaItem } = data;
+  
+  const [isPlayingArray, setIsPlayingArray] = useState(
+    mediaItem.map(() => false)
+  );
+
+  const handlePlayVideo = (index) => {
+    const newIsPlayingArray = [...isPlayingArray];
+    newIsPlayingArray[index] = true;
+    setIsPlayingArray(newIsPlayingArray);
+  };
+
+  const renderMedia = (media, index) => {
+    const isPlaying = isPlayingArray[index];
+
+    const sizeClasses = "w-[500px] h-[300px] sm:w-[500px] sm:h-[300px] md:w-[500px] md:h-[320px] lg:w-[600px] lg:h-[340px] xl:w-[600px] xl:h-[340px]";
+
+    if (
+      media.__typename ===
+      'PageBlocksMediaComponentMediaItemCloudinaryMediaComponent'
```

Comment:
Any ideas to make it so these are not hardcoded?

Reply:
not within the media data object